### DISCUSSION TITLE
how to uninstall plugins not to break site

### DISCUSSION THREAD

**Discussion by danie-dejager** [netbox-community/netbox#10641] [Q&A] [answered]
Hi,

What are the steps to remove / uninstall plugins? I run 3.3.4 and when I removed the installed plugin from configuration.py I got an error page to Netbox and could no longer access it. I had to put the plugin back in the config.

TIA

**Comment by bogdancordos** [comment]
I think you need to remove the plugin from _configuration.py_, as you did, and from _local_requirements.txt_ then do a _./upgrade.sh_

**Comment by danie-dejager** [comment]
That's exactly what I did but thereafter the Netbox frontend was no longer available. I had to add the plugin back.

**Reply by bogdancordos** [reply]
What plugin are we talking about, if I may ask? And what error you get after uninstall?

**Reply by danie-dejager** [reply]
It is the `Netbox-DNS` plugin specifically but when I remove the `netbox_secretstore` plugin I get the same issue.

Here are my steps:
1. vi configuration.py and remove plugins. Currently it looks like this
`PLUGINS = ["netbox_secretstore","netbox_topology_views"]`
2. remove plugin from local_requirements.txt
```
→ cat local_requirements.txt
netbox-topology-views
netbox-secretstore
#netbox-dns
social-auth-app-django
```
4. ran upgrade
`PYTHON=/usr/bin/python3.8 /opt/netbox/upgrade.sh`
5.  restarted
`systemctl restart netbox netbox-rq`
 
Server Error:
There was a problem with your request. Please contact an administrator.

The complete exception is provided below:

<class 'TypeError'>

__init__() missing 1 required positional argument: 'strategy'

Python version: 3.8.5
NetBox version: 3.3.5

![image](https://user-images.githubusercontent.com/13314239/195800534-e79f7cbe-0434-4919-92cb-d58b9a4e3ddf.png)

When I return the plugin 
`PLUGINS = ["netbox_secretstore","netbox_topology_views","netbox_dns"]`
and add it back to local_requirements.txt the site works again

**Comment by danie-dejager** [comment, marked as answer]
I found that as I had an older install that I kept upgrading and by using Azure AD I had a slightly different default config than the latest version. Somehow the social-auth plugin cause all this confusion. Once I reconfigured `configuration.py` from scratch and only adding back what is needed I no longer had this issue. 

The issue was never that the plugins being removed broke the Netbox UI but rather an incorrect configuration to make AzureAD SSO work.
